### ISSUE TITLE
qt/5.x.x: Requires main even for shared lib

### ISSUE THREAD

**Issue by Fettpet** [conan-io/conan-center-index#7914]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: qt/5.15.2
  * Operating System+version: Windows
  * Compiler+version: Visual Studio
  * Docker image: every
  * Conan version: every
  * Python version: every

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)

[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce (Include if Applicable)
Setup a project to build a dll. The dll depends on qt5. The dll has no main function. Because of line [882 + 883](https://github.com/conan-io/conan-center-index/blob/b4c5601c6b0f3cc0b6382cbcf3a0cb22452426a0/recipes/qt/5.x.x/conanfile.py#L882) each artefact needs a main function. 

### Expectations
I can include the qt5 recept without the requirements to have a main function. This approach would make the workaround of [QWT](https://github.com/conan-io/conan-center-index/pull/7232)  useless.

**Comment by ericLemanissier** [comment]
I don't understand which problem you have currently. Could you please send a full log of a failure, which the conanfile and cmakelists.txt files used ?

**Comment by perseoGI** [comment]
Hi there, 
I was reviewing old issues and I have come across this one.

I'm closing this issue as this recipe has suffer plenty of modifications since this issue was reported.
There is not anymore a QT `ENTRYPOINT` on this recipe.

If for some reason the problem remains nowadays, do not hesitate to reopen this issue or create a new ticket with updated information, preferably using `conan v2`.

Happy coding 🐸
